### PR TITLE
[App Service] BREAKING CHANGE: `az webapp up`: Use deployment status API by default for Linux Webapp Deployments

### PR DESCRIPTION
**Pull request by sarsharma**
This is continuation of changes in https://github.com/Azure/azure-cli/pull/28235

**Related command**

az webapp deploy
az webapp up
az webapp deployment source config-zip

**Description**

We had introduced a feature flag `--track-status` for webapp deployment commands, which enabled the use of Deployment Status API while deploying Linux Web Apps. With this change, this will become the default deployment experience for Linux Webapp Deployments. Users can still switch to older deployment output experience by using `--track-status false`  

**Testing Guide**

Deploying webapps using CLI with command `az webapp up`, `az webapp deployment source-config` and `az webapp deploy`

* `az webapp deploy` without `--track-status` arg

image

 * `az webapp deploy` with `--track-status false` arg
image

**History Notes**

[App Service] BREAKING CHANGE: `az webapp deploy`: Use deployment status API for deployment output for Linux Web Apps
[App Service] BREAKING CHANGE: `az webapp up`: Use deployment status API for deployment output for Linux Web Apps
[App Service] BREAKING CHANGE: `az webapp deployment source config-zip`: Use deployment status API for deployment output for Linux Web Apps

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Please fix CI issues

**Review by ShilpiRachna1 — APPROVED** [review]
Looks good 👍

**Review by CodingIsBliss — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -676,7 +676,7 @@ def enable_zip_deploy_functionapp(cmd, resource_group_name, name, src, build_rem
     return enable_zip_deploy(cmd, resource_group_name, name, src, timeout, slot)
 
 
-def enable_zip_deploy_webapp(cmd, resource_group_name, name, src, timeout=None, slot=None, track_status=False):
+def enable_zip_deploy_webapp(cmd, resource_group_name, name, src, timeout=None, slot=None, track_status=True):
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:679]: I would like to confirm if changing this default value will cause a breaking change in the customer's automation workflow?

Comment (sarsharma) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:679]: @zhoxing-ms Yes, this will be a breaking change.

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**